fastlane/Fastfile

```diff
@@ -0,0 +1,85 @@
+RELEASE_TAG_PREFIX = 'release_v'
+default_platform(:android)
+
+platform :android do
+  desc "Create a pre-release version by pushing a tag to GitHub, and publishing to Maven"
+  lane :unstable do 
+    next_version = calculate_next_canary_version({
+        :release_tag_prefix => RELEASE_TAG_PREFIX,
+        :release_tag_type => 'annotated'
+      }
+    )
+
+    UI.message("Releasing unstable version: #{next_version}")
+
+    # Update version information in gradle.properties
+    increment_versions(version: next_version)
+
+    gradle(
+      task: "assemble",
+      build_type: "Release"
+    )
+
+    # # Create tag and push to origin
+    add_tag(version: RELEASE_TAG_PREFIX + next_version.to_s)
+
+    # # Push to maven
+    UI.message("Pushing #{next_version.to_s} to maven")
+    publish_to_maven
```

Comment:
Is the idea here that the different lanes have different publications? private -> local, public -> maven central?

What's the use case for Maven local? Would we run the Fastlane script on our local machines?

Reply:
Maven local was just a placeholder. I wanted to get the bits in place up to the point where I'd be publishing to maven. There will be a follow-up PR for that with the changes to publish to sonatype.

Regarding the lanes, my understanding is that "public" lanes (in our case unstable and release) are the only ones that can be invoked when using the fastlane cli. `bundle exec fastlane <platform> <public lane name>`. "Private" lanes cannot be invoked directly and basically serve as a way to encapsulate some functionality you might use from different public lanes. 

Commits to the `main` branch will trigger a step in the circleci workflow (after all the testing) which will run the `unstable` lane: `bundle exec fastlane android unstable`. The artifacts of a successful build will then be published to maven with version `1.1.2-unstable.0` and a tag will be created with `release_v1.1.2-unstable.0`.

The release lane is similar, but will also generate release notes and will be published to maven as the final release with version `1.1.2` and tagged as `release_v1.1.2`. I'm not sure what the trigger will be (iOS is using a release branch). The first step is to get unstable builds published so developers can use those unstable builds to verify fixes and things of that sort.